app/models/challenge_models/gift_exchange.rb

```diff
@@ -80,4 +80,21 @@ def user_allowed_to_see_requests_summary?(user)
   def user_allowed_to_see_prompt?(user, prompt)
     self.collection.user_is_maintainer?(user) || prompt.pseud.user == user
   end
+
+  # reindex collection after creation, deletion, and certain attribute updates
+  after_create :reindex_collection
+  after_destroy :reindex_collection
+  after_update :reindex_collection, if: Proc.new { |c| c.saved_change_to_signup_open? ||
```

Comment:
You can write the `saved_change_` checks like this, where you provide a list of attributes (`id` replaces `after_create`):

https://github.com/otwcode/otwarchive/blob/f057b6b9f85ccdd26dcff2aca2efa17cf7f38e26/app/models/pseud.rb#L478-L486